### PR TITLE
Add (minimal) support for VBMDALI

### PR DESCRIPTION
**Pull request by niobos**
This PR is an attempt to add (minimal) support for the VBMDALI module. This is currently limited to LedModule DALI devices, since these are the only ones I have access to.

Relates to #32

### CODE REVIEW

**Review by Cereal2nd — APPROVED** [review]
i know its early, but this makes the reviewing simpler if i can do it in smaller parts.

For my knowledge (i'm not a great python programmer) why the module factory instead of a new module?

**Comment by niobos** [comment]
Thank you for the approval so far. That way I'm confident that I'm not going in a direction you don't want.

I may be rebasing and reordering these commits, though. I'm not sure what will happen to your preliminary approval.

The factory is so that I can do something like this:

```python
class Module:
    @classmethod
    def factory(cls, module_type) -> Module:
        if module_type == "VMBDALI":
            return VmbDali(module_type)
        # else:
        return Module(module_type)
    ...
    def some_method(self):
        ...  # default behaviour

class VmbDali(Module):
    """Behaves exactly like Module, but allows module-specific overrides"""
    ...
    def some_method(self):
        ...  # different behaviour vs Module
```

**Comment by Cereal2nd** [comment]
this commit is becoming so big it becomes dangerous to break things
as we have no testing for the velbusaio lib on its own it hink we may need to merge this one in smaller chunks
certainly the "general" areas that are touched

**Comment by niobos** [comment]
I tried to keep each individual commit very contained, but indeed, the whole PR is getting big...
What parts would you like to merge first?

**Comment by Cereal2nd** [comment]
for example, first all the messaging changes,
then the controller/module changes

this will give us the possibility to test this before we add another change on top.
I just want to be sure we do not destabalize the system, if we had automated test for the core system this would be less of a concern.

Maybe its time to start adding testcases ...

**Comment by niobos** [comment]
I've rebased this PR on top of the new master (after #49). It now only contains module/channel changes, separated in 3 commits for easier reviewing.

File: velbusaio/module.py [hunk 1]
```diff
@@ -582,3 +626,136 @@ async def __load_default_channels(self) -> None:
...
+        await self._writer(msg)
+
+    async def on_message(self, message: Message) -> None:
+        if isinstance(message, DaliDeviceSettingMsg):
+            if isinstance(message.data, DaliDeviceTypeMsg):
+                if message.data.device_type == DaliDeviceType.NoDevicePresent:
+                    if message.channel in self._channels:
+                        del self._channels[message.channel]
+                elif message.data.device_type == DaliDeviceType.LedModule:
+                    if self._channels.get(message.channel).__class__ != Dimmer:
```
Comment (Cereal2nd) [inline velbusaio/module.py:691]: how come we do not know this is a dimmer channel up front?

File: velbusaio/messages/__init__.py [hunk 2]
```diff
@@ -36,7 +36,7 @@
 from velbusaio.messages.cover_off import CoverOffMessage, CoverOffMessage2
 from velbusaio.messages.cover_position import CoverPosMessage
 from velbusaio.messages.cover_up import CoverUpMessage, CoverUpMessage2
-from velbusaio.messages.dali_device_settings import DaliDeviceSettings
+from velbusaio.messages.dali_device_settings import DaliDeviceSettingMsg
```
Comment (Cereal2nd) [inline velbusaio/messages/__init__.py:39]: This should have been in the previous PR, i fixed it manually in the master branch, please rebase

File: velbusaio/module.py (hunk 1 above)
Comment (niobos) [reply velbusaio/module.py:691]: DALI also supports other types of devices, such as Relays. It's probably better to expose these as a Relay channel. This code ensures our local view of a channel matches with the DALI-bus-side.

I only have "LedModule" DALI devices, so I can't check/test the other behavior.

**Comment by niobos** [comment]
Thank you for reviewing this and accepting this PR!

**Pull request merged**